### PR TITLE
API changes

### PR DESCRIPTION
**Pull request by abraae**
New/updated APIs to create/update jobs:
- POST /jobs
- PATCH /jobs/byID/{job}

New APIs to deal with users (who may be recruiters or managers), including concept of deactivation:
- POST /users
- GET /users/byID/{id}
- PATCH /users/byID/{id}
- GET /users/active
- GET /users/active/byID/{id}
- GET /users/active/managers
- GET /users/active/managers/byID/{id}
- GET /users/active/recruiters
- GET /users/active/recruiters/byID/{id}

New/updated APIs for bulk operations on category values:
- POST /categories/byID/{category}/values/operations/patches 
- POST /categories/byID/{category}/values/operations/replaces
- GET /categories/byID/{category}/values/operations/byID/{operationID}

### CODE REVIEW

**Comment by abraae** [comment]
In progress - don't merge yet.

File: docs/raml/recruitment.raml [hunk 1]
```diff
@@ -1122,12 +1122,31 @@ resourceTypes: !include resources.yml
...
+              get:
+                securedBy: [oauth_2_0]
+                description: |
+                  Get the status of a previously started patch or replace operation. Normally the producer should keep results around
+                  for some time (i.e. weeks/months) after the operation has completed.
+         
+                  SoT.
+                responses:
+                  200:
+                    description: the category bulk operation completed successfully
```
Comment (bgorven) [inline docs/raml/recruitment.raml]: @abraae how would we respond when the operation fails?

**Comment by bgorven** [comment]
@abraae the jobWrite schema appears to be missing the 'workflow' and 'description' properties that are present on jobRead.

**Comment by abraae** [comment]
Sorry won't be bear a computer until Saturday, will fix then

**Pull request merged**